app/src/hooks/initialize-agent.ts

```diff
@@ -0,0 +1,262 @@
+import { Agent, HttpOutboundTransport, MediatorPickupStrategy, WsOutboundTransport, WalletError } from '@credo-ts/core'
+import { IndyVdrPoolConfig, IndyVdrPoolService } from '@credo-ts/indy-vdr/build/pool'
+import { useAgent } from '@credo-ts/react-hooks'
+import { agentDependencies } from '@credo-ts/react-native'
+import {
+  BifoldError,
+  DispatchAction,
+  useAuth,
+  useStore,
+  migrateToAskar,
+  createLinkSecretIfRequired,
+  TOKENS,
+  useServices,
+  PersistentStorage,
+} from '@hyperledger/aries-bifold-core'
+import { GetCredentialDefinitionRequest, GetSchemaRequest } from '@hyperledger/indy-vdr-shared'
+import moment from 'moment'
+import { useCallback } from 'react'
+import { Config } from 'react-native-config'
+import { CachesDirectoryPath } from 'react-native-fs'
+import { activate } from '../helpers/PushNotificationsHelper'
+import { getBCAgentModules } from '../helpers/bc-agent-modules'
+import { BCState, BCLocalStorageKeys } from '../store'
+
+const loadCachedLedgers = async (): Promise<IndyVdrPoolConfig[] | undefined> => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const cachedTransactions = await PersistentStorage.fetchValueForKey<any>(BCLocalStorageKeys.GenesisTransactions)
+  if (cachedTransactions) {
+    const { timestamp, transactions } = cachedTransactions
+    return moment().diff(moment(timestamp), 'days') >= 1 ? undefined : transactions
```

Comment:
1 might be a magic number here. maxAgeInDays or cacheAgeInDays?

Reply:
This was copied exactly as is from the splash screen in the spirit of keeping all the logic the same but I can make this change anyway